File: app/src/main/java/com/antonio/samir/meteoritelandingsspots/ui/recyclerView/selector/MeteoriteSelectorView.java
package com.antonio.samir.meteoritelandingsspots.ui.recyclerView.selector;

import com.antonio.samir.meteoritelandingsspots.model.Meteorite;

/**
 * Interface used by view activity to implement proper commands for each meteorite selection case
 */
public interface MeteoriteSelectorView {

    /**
     * Exceute landscape command for meteorite selection
     * @param meteorite
     */
    void selectLandscape(final Meteorite meteorite);

    /**
     * Exceute portrait command for meteorite selection
     * @param meteorite
     */
    void selectPortrait(final Meteorite meteorite);

}

File: app/src/main/java/com/antonio/samir/meteoritelandingsspots/features/list/recyclerView/selector/MeteoriteSelectorView.kt
package com.antonio.samir.meteoritelandingsspots.features.list.recyclerView.selector

import com.antonio.samir.meteoritelandingsspots.data.repository.model.Meteorite

/**
 * Interface used by view activity to implement proper commands for each meteorite selection case
 */
interface MeteoriteSelectorView {

    /**
     * Exceute landscape command for meteorite selection
     * @param meteorite
     */
    fun selectLandscape(meteorite: Meteorite)

    /**
     * Exceute portrait command for meteorite selection
     * @param meteorite
     */
    fun selectPortrait(meteorite: Meteorite)

}
